rfcs/0012-external-function.md

```diff
@@ -0,0 +1,163 @@
+---
+feature: external-function-framework
+authors:
+  - "TennyZhuang"
+start_date: "2022/11/04"
+---
+
+# External Function Framework
+
+## Summary
+
+We will introduce a gRPC-based external function framework in the kernel part. **The framework is private to users and may change later**.
```

Comment:
If we want to be compatible with arrow, why not use arrow flight rpc? https://arrow.apache.org/docs/format/Flight.html#

Reply:
We don't use an arrow format library and write it manually, so it's hard for us to use a flight library directly, and we don't want to waste time for that.

Reply:
Anyway, it's acceptable to me, but I don't want to investigate it now.